### PR TITLE
Fix various missing deps and not needed coupling to Symfony

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT
| Doc PR          | -

### CODE REVIEW

File: src/Sylius/Bundle/ThemeBundle/composer.json [hunk 1]
```diff
@@ -36,14 +36,16 @@
         "phpunit/phpunit": "^4.1",
         "symfony/twig-bundle": "^2.7.7"
     },
-    "minimum-stability": "dev",
-    "prefer-stable": true,
```
Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/composer.json]: Why?

File: src/Sylius/Component/Metadata/composer.json [hunk 2]
```diff
@@ -20,10 +20,19 @@
...
+
+        "sylius/resource": "0.17.*",
+        "symfony/property-access": "^2.7",
+        "symfony/options-resolver": "^2.7",
+        "twig/twig": "~1.11"
     },
     "require-dev": {
-        "phpspec/phpspec": "^2.4"
+        "phpspec/phpspec": "^2.4",
+        "phpunit/phpunit": "^4.1"
```
Comment (pamil) [inline src/Sylius/Component/Metadata/composer.json]: We don't use PhpUnit in Metadata component IIRC.

File: src/Sylius/Bundle/ThemeBundle/composer.json (hunk 1 above)
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ThemeBundle/composer.json]: We do not have ``prefer-stable`` in other bundles and Symfony doesn't use it - https://github.com/symfony/http-foundation/blob/master/composer.json. If we decide it is good idea to add it then we should do it for all bundles and components.

**Comment by michalmarcinkowski** [comment]
:+1:

File: src/Sylius/Bundle/ThemeBundle/composer.json (hunk 1 above)
Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/composer.json]: It's definitely a good idea, without prefer-stable we will install unstable versions of all our dependencies, whereas stable versions suits as well. We should install dev versions only if there isn't any other version available.

File: src/Sylius/Component/Metadata/composer.json (hunk 2 above)
Comment (pjedrzejewski) [inline src/Sylius/Component/Metadata/composer.json]: You used assert classes.

File: src/Sylius/Bundle/ThemeBundle/composer.json (hunk 1 above)
Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/composer.json]: Btw. that Symfony package doesn't have `.travis.yml` too, it's meant to be tested as a part of `symfony/symfony`.

File: src/Sylius/Component/Metadata/composer.json (hunk 2 above)
Comment (pamil) [inline src/Sylius/Component/Metadata/composer.json]: :+1:, should be updated to `expect` in the future :)

File: src/Sylius/Bundle/ThemeBundle/composer.json (hunk 1 above)
Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/composer.json]: As an example: https://travis-ci.org/Sylius/SyliusResourceBundle/jobs/111649988

```
- Installing symfony/twig-bridge (3.0.x-dev 2a46495)
  Cloning 2a4649506e9e538b6f42b5f1c7545fca99ff6e5a
- Installing symfony/twig-bundle (2.8.x-dev 0901cdf)
  Cloning 0901cdfd83379cedde8cd57a24dd44969c0a38a2
```

Twig bridge `3.0.x-dev` mixed up with twig bundle `2.8.x-dev`, what could possibly go wrong? :)

**Pull request merged**